apps/site/pages/en/learn/asynchronous-work/discover-promises-in-nodejs.md

```diff
@@ -0,0 +1,317 @@
+
+```js
+Promise.resolve('Resolved immediately').then(result => {
+  console.log(result); // 'Resolved immediately'
+});
+```
+
+## Error Handling with Promises
+
+Handling errors in Promises ensures your application behaves correctly in case of unexpected situations.
+
+- You can use `.catch()` to handle any errors or rejections that occur during the execution of Promises.
+
+```js
+myPromise
+  .then(result => console.log(result))
+  .catch(error => console.error(error)); // Handles the rejection
+```
+
+- Alternatively, when using `async/await`, you can use a `try/catch` block to catch and handle errors.
+
+```js
+async function performTask() {
+  try {
+    const result = await myPromise;
+    console.log(result);
+  } catch (error) {
+    console.error(error); // Handles any errors
+  } finally {
+    // This code is executed regardless of failure
+    console.log('performTask() completed');
+  }
+}
+
+performTask();
+```
+
+## Scheduling Tasks in the Event Loop
+
+In addition to Promises, Node.js provides several other mechanisms for scheduling tasks in the event loop.
+
+### **`queueMicrotask()`**
+
+`queueMicrotask()` is used to schedule a microtask, which is a lightweight task that runs after the currently executing script but before any other I/O events or timers. Microtasks include tasks like Promise resolutions and other asynchronous operations that are prioritized over regular tasks.
+
+```js
+queueMicrotask(() => {
+  console.log('Microtask is executed');
+});
+
+console.log('Synchronous task is executed');
+```
+
+In the above example, "Microtask is executed" will be logged after "Synchronous task is executed," but before any I/O operations like timers.
+
+### **`process.nextTick()`**
+
+`process.nextTick()` is used to schedule a callback to be executed after the current operation completes, but before any other event loop phases, such as I/O events, timers, or Promises. This is useful for situations where you want to ensure that a callback is executed as soon as possible, but still after the current execution context.
+
+```js
+process.nextTick(() => {
+  console.log('Next tick callback');
+});
+
+console.log('Synchronous task executed');
+```
+
+### **`setImmediate()`**:
+
+`setImmediate()` is used to execute a callback after the current event loop cycle finishes and all I/O events have been processed. This means that `setImmediate()` callbacks run after any I/O callbacks, but before timers and the next tick queue.
+
+```js
+setImmediate(() => {
+  console.log('Immediate callback');
+});
+
+console.log('Synchronous task executed');
+```
+
+### **When to Use Each**:
+
+- Use `queueMicrotask()` for tasks that need to run immediately after the current script and before any I/O or timer callbacks, typically for Promise resolutions.
+- Use `process.nextTick()` for tasks that should execute before any I/O events, often useful for deferring operations or handling errors synchronously.
```

Comment:
It's likely worth pointing out that error handling between `process.nextTick(...)` and `queueMicrotask(...)` is quite different.

Reply:
How so?

Reply:
Just for example.. 
![image](https://github.com/user-attachments/assets/e6dc5142-dbb3-4330-aa06-0bf4774b62cc)